packages/menu/MenuListItem.tsx

```diff
@@ -0,0 +1,54 @@
+// The MIT License
+//
+// Copyright (c) 2019 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as React from 'react';
+import {ListItem, ListItemProps} from '@material/react-list';
+
+export interface MenuListItemProps<T> extends ListItemProps<T> {
+  children?: React.ReactNode;
+}
+
+class MenuListItem<T extends HTMLElement = HTMLElement> extends React.Component<MenuListItemProps<T>, {}> {
```

Comment:
this could be a functional component

Reply:
I actually need this to be a Compnonent to pass along the defaultProps. Any idea around that?

Reply:
yeah, I don't think we could reuse the default props from the list item :-/

Reply:
The defaultProp is the `renderWithListItemProps`, so it can't live on the listItem. Is that what you are suggesting?

Reply:
no, my idea would be to use default parameters on the function (which is the recommended way to use default props on function component) like this

```tsx
const MenuListItem = <T extends HTMLElement = HTMLElement>({
  renderWithListItemProps = true,
  ...props
}) => {
  // etc
}
```

but then we would need to redeclare every prop on the `defaultProps` of the `ListItem`. does that make sense?

Reply:
I tested it using this as the component: 
```
const MenuListItem: <T extends HTMLElement = HTMLElement>(props: MenuListItemProps<T>)
=> React.ReactElement<MenuListItemProps<T>> = ({
  renderWithListItemProps = true,
  role = 'menuitem',
  children,
  checkboxList = false,
  radioList = false,
  className = '',
  tabIndex = -1,
  onKeyDown = () => {},
  onClick = () => {},
  onFocus = () => {},
  onBlur = () => {},
  tag = 'li',
  ...props
}) => {

  return (
    <ListItem
      role={role}
      {...props}
    >
      {children}
    </ListItem>
  );
}

```

This actually doesn't carry the defaultProp to the ListItem. Is that what you were thinking?

Reply:
exactly, but then we have the duplication of code (one in the default parameters of the `MenuListItem` and other on the `defaultProps` of the `ListItem`), but if you are okay with that then it's all good :-)

Reply:
It actually doesn't work as we think it would. So its not a viable solution.

Reply:
oh, I missed this, but shouldn't you also pass along the props defined? Like
```tsx
const MenuListItem: <T extends HTMLElement = HTMLElement>(props: MenuListItemProps<T>)
=> React.ReactElement<MenuListItemProps<T>> = ({
  renderWithListItemProps = true,
  role = 'menuitem',
  children,
  checkboxList = false,
  radioList = false,
  className = '',
  tabIndex = -1,
  onKeyDown = () => {},
  onClick = () => {},
  onFocus = () => {},
  onBlur = () => {},
  tag = 'li',
  ...props
}) => {

  return (
    <ListItem
      role={role}
      renderWithListItemProps={renderWithListItemProps}
      children={children}
      checkboxList={checkboxList}
      radioList={radioList}
      className={className}
      tabIndex={tabIndex}
      onKeyDown={onKeyDown}
      onClick={onClick}
      onFocus={onFocus}
      onBlur={onBlur}
      tag={tag}
      {...props}
    >
      {children}
    </ListItem>
  );
}
```

Reply:
or maybe passing on only what we want to "override" from the default props

```tsx
const MenuListItem: <T extends HTMLElement = HTMLElement>(props: MenuListItemProps<T>)
=> React.ReactElement<MenuListItemProps<T>> = ({
  renderWithListItemProps = true,
  role = 'menuitem',
  ...props
}) => {

  return (
    <ListItem
      role={role}
      renderWithListItemProps={renderWithListItemProps}
      {...props}
    >
      {children}
    </ListItem>
  );
}
```

Reply:
sorry I think this ended up being confusing 😅

but I believe this last snippet should achieve the correct behaviour

Reply:
ah ya - of course I need to pass that along...however, I don't think we need to pass the same defaultProps to the ListItem...The reason I have `renderWithListItemProps` on the MenuListItem, is so that the List knows to pass along certain props to the MenuListItem. Does that make sense?

Reply:
oh, I see. I think that's kind a weird way to do it :smile:

can you add to the check of `element.type === ListItem` the `MenuListItem`?

Reply:
Ya I realize that's weird...but it also seemed wierd that MenuListItem is referenced in List. What do you think?

Reply:
I guess that is the better more foolproof way :| -- I'll wait to hear back from you if you think of something better.

Reply:
you can create a context in the `List` component and use the context consumer in both `ListItem` and `MenuListItem`, and in the context you can put the functions you need in the item. What do you think?